From 525cf94a53c353755e04a54994f394b54c23fcb2 Mon Sep 17 00:00:00 2001
From: Alwin Esch <alwin.esch@web.de>
Date: Sun, 25 Sep 2016 10:35:47 +0200
Subject: [PATCH] [paplayer] fix codec deinit (was never called)

It calls DeInit now on ADDON::CAudioDecoder like on DVDPlayerCodec.
Also becomes the DeInit removed from ICodec since it is handled on the
classes themselves.
---
 xbmc/addons/AudioDecoder.cpp           | 5 +++++
 xbmc/addons/AudioDecoder.h             | 2 +-
 xbmc/cores/paplayer/ICodec.h           | 5 -----
 xbmc/cores/paplayer/DVDPlayerCodec.h | 2 +-
 4 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/xbmc/addons/AudioDecoder.cpp b/xbmc/addons/AudioDecoder.cpp
index 6625898..8ab0d2d 100644
--- a/xbmc/addons/AudioDecoder.cpp
+++ b/xbmc/addons/AudioDecoder.cpp
@@ -41,6 +41,11 @@
   // Copy constructor is generated by compiler and calls parent copy constructor
   return AddonPtr(new CAudioDecoder(*this));
 }
+
+CAudioDecoder::~CAudioDecoder()
+ {
+   DeInit();
+ }
 
 bool CAudioDecoder::Init(const std::string& strFile, unsigned int filecache)
 {
diff --git a/xbmc/addons/AudioDecoder.h b/xbmc/addons/AudioDecoder.h
index 493830e..e3fc976 100644
--- a/xbmc/addons/AudioDecoder.h
+++ b/xbmc/addons/AudioDecoder.h
@@ -50,7 +50,7 @@
       , m_channel{nullptr}
     {};
     CAudioDecoder(const cp_extension_t *ext);
-    virtual ~CAudioDecoder() {}
+    virtual ~CAudioDecoder();
     virtual AddonPtr Clone() const;
 
     // Things that MUST be supplied by the child classes
diff --git a/xbmc/cores/paplayer/ICodec.h b/xbmc/cores/paplayer/ICodec.h
index baae3d0..7c7da32 100644
--- a/xbmc/cores/paplayer/ICodec.h
+++ b/xbmc/cores/paplayer/ICodec.h
@@ -56,11 +56,6 @@ class ICodec
   // 3.  Fill in the m_TotalTime, m_SampleRate, m_BitsPerSample and m_Channels parameters.
   virtual bool Init(const CFileItem &file, unsigned int filecache)=0;
 
-  // DeInit()
-  // Should just cleanup anything as necessary.  No need to free buffers here if they
-  // are allocated and destroyed in the destructor.
-  virtual void DeInit()=0;
-
   virtual bool CanSeek() {return true;}
 
   // Seek()
